packages/patternfly-react/src/components/SessionTimeout/helpers.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export class Session extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      now: this.props.now(),
+      expiresAt: this.props.expiresAt
+    };
+  }
+
+  componentWillMount() {
+    this._interval = setInterval(
+      () => this.setState(() => ({ now: this.props.now() })),
+      1000
+    );
+  }
+
+  componentWillUnmount() {
+    clearInterval(this._interval);
+  }
+
+  render() {
+    const logout = () => {
+      clearInterval(this._interval);
+      this.setState(() => ({ expiresAt: 0 }));
+    };
+    const contin = () =>
+      this.setState(() => ({
+        expiresAt: this.props.now() + this.props.sessionTime,
+        now: this.props.now()
+      }));
+
+    if (!this._interval && this.state.expiresAt - this.state.now > 0) {
```

Comment:
I don't think this will ever be hit. As well this is also a side effect in render and should be avoided.